File: src/main/java/org/capstone/backend/service/vote/EvidenceVoteService.java
package org.capstone.backend.service.vote;


import org.capstone.backend.dto.evidence.EvidenceVoteHistoryResponse;
import org.capstone.backend.dto.vote.EvidenceVoteRequest;
import org.capstone.backend.dto.vote.EvidenceVoteResponse;

import java.util.List;

public interface EvidenceVoteService {
    String voteEvidence(Long evidenceId, EvidenceVoteRequest request);
    List<EvidenceVoteResponse> getEvidenceToVoteByChallenge(Long challengeId);
    List<EvidenceVoteResponse> getVoteHistoryForChallenge(Long challengeId);
    List<EvidenceVoteHistoryResponse> getMyVotedEvidence(Long challengeId);
}

File: frontend/src/service/evidenceVoteService.js
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { BASE_URL } from "../contant/contant.js";

export const evidenceVoteService = createApi({
    reducerPath: "evidenceVote",
    baseQuery: fetchBaseQuery({
        baseUrl: BASE_URL,
        prepareHeaders: (headers) => {
            const token = localStorage.getItem("jwt_token");
            if (token) {
                headers.set("Authorization", `Bearer ${token}`);
            }
            headers.set("Content-Type", "application/json");
            return headers;
        },
    }),
    tagTypes: ["EvidenceVote"],
    endpoints: (builder) => ({
        // ✅ Vote on evidence
        voteEvidence: builder.mutation({
            query: ({ evidenceId, score }) => ({
                url: `/evidence-votes/${evidenceId}`,
                method: "POST",
                body: { score },
            }),
            invalidatesTags: ["EvidenceVote"],
        }),

        // ✅ Get evidence list to vote by challenge
        getEvidenceToVoteByChallenge: builder.query({
            query: (challengeId) => `/evidence-votes/tasks/${challengeId}`,
            providesTags: ["EvidenceVote"],
        }),

        // ✅ Add to endpoints:
        getMyVotedEvidence: builder.query({
            query: (challengeId) => ({
                url: "/evidence-votes/my-votes",
                method: "GET",
                params: challengeId ? { challengeId } : undefined,
            }),
            providesTags: ["EvidenceVote"],
        }),
    }),
});

export const {
    useVoteEvidenceMutation,
    useGetEvidenceToVoteByChallengeQuery,
    useGetMyVotedEvidenceQuery,
} = evidenceVoteService;
